src/tokenizer/tokenizer.py

```diff
@@ -1539,13 +1540,18 @@ def parse_tokens(txt: Union[str, Iterable[str]], **options: Any) -> Iterator[Tok
                     # Probably an idiot trying to type opening double quotes with commas
                     punct, rt = rt.split(2)
                     yield TOK.Punctuation(punct, normalized="„")
-                elif lw >= 2 and (rtxt.startswith("[[") or rtxt.startswith("]]")):
+                elif lw >= 2 and (
+                    (rtxt.startswith("[[") and not inside_paragraph_marker)
+                    or (rtxt.startswith("]]") and inside_paragraph_marker)
+                ):
                     # Begin or end paragraph marker
                     marker, rt = rt.split(2)
-                    if marker.txt == "[[":
+                    if marker.txt == "[[" and not inside_paragraph_marker:
```

Comment:
`and not inside_paragraph_marker` can be omitted - it is always true because of the enclosing elif condition.

Reply:
Fixed!